source/event_stream.c

```diff
@@ -689,14 +691,27 @@ struct aws_event_stream_header_value_pair aws_event_stream_create_int32_header(
     return header;
 }
 
+static int s_copy_header_name(struct aws_event_stream_header_value_pair *header, void *name, size_t name_len) {
+
+    if (name_len >= sizeof(header->header_name)) {
+        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
```

Comment:
`OVERFLOW_DETECTED` is kinda vague
we probably want to define `AWS_EVENT_STREAM_MAX_HEADER_NAME_SIZE` and `AWS_EVENT_STREAM_MAX_HEADER_VALUE_SIZE` and add specific errors for it

event_stream.h already defines
```
/* max message size is 16MB */
#define AWS_EVENT_STREAM_MAX_MESSAGE_SIZE (16 * 1024 * 1024)

/* max header size is 128kb */
#define AWS_EVENT_STREAM_MAX_HEADERS_SIZE (128 * 1024)
```
and has well-named errors like:
```
    AWS_ERROR_EVENT_STREAM_MESSAGE_FIELD_SIZE_EXCEEDED,
```